Scripts/Player/Player.gd

```diff
@@ -15,32 +15,37 @@ var path: Array = []                        ## Array to hold the calculated path
 var path_index: int = 0                     ## Current index in the path array.
 var highlight_path: PackedVector2Array = [] ## Array to hold the positions to highlight on the path.
 var selected_object: InteractableObject     ## Reference to the interactable object currently selected by the player.
+var camera: Camera2D                        ## Refrence to Camera
 var action: bool = false:                   ## Flag indicating if the player is currently performing an action.
 	set(value):
 		action = value
 		queue_redraw()
 
+# --- Private Properties ---
+@onready var _actions: int = actions
+
 # --- Signals ---
 signal moved(position: Vector2i)            ## Signal emitted when the player’s position is updated.
-
+signal turn_ended                           ## Signal emitted when the player's turn ends
 # --- Built-in Callbacks ---
 # Called when the node is added to the scene. Sets up camera limits and connects the resize signal.
 func _ready() -> void:
-	
+	camera = get_child(UtilityFunctions.get_from_type(get_children(), "Camera2D"))
```

Comment:
Instead may be more prudent to simply give an @onready variable for the camera, or 
`for i in get_children():
    if i is Camera2D:
          camera = i`
This is mostly unnecessary.